### ISSUE TITLE
Support for Html.Keyed style containers

### ISSUE THREAD

**Issue by norpan** [rundis/elm-bootstrap#27]
There is currently no support for `Html.Keyed` style containers.

`Html.Keyed` is needed to optimize DOM updates when children are moved around.

In some cases, for instance `Grid.col`, you can just add a child `Html.Keyed.node "div" []` but in others, for instance `Grid.row`, you cannot, because it only accepts `Grid.col` children.

I propose making a `Bootstrap.Grid.Keyed` (or `Bootstrap.Keyed.Grid`, or just `BootStrap.Keyed`) module that has
```
row : List (Option msg) -> List (String, Column msg) -> Html msg
col : List (Option msg) -> List (String, Html msg) -> Column msg
```
or as an alternative, put the functions in the `Bootstrap.Grid` module as
```
rowKeyed/keyedRow : List (Option msg) -> List (String, Column msg) -> Html msg
colKeyed/keyedCol : List (Option msg) -> List (String, Html msg) -> Column msg
```
and the same for other places where there are containers, like `Table`where you would get a large DOM update if you move a row to the top unless it's keyed. I'm not sure where you would actually need keyed children, but I guess you cannot really predict real world usage.

I can make a PR if you think this is a good idea.

**Comment by rundis** [comment]
I opted for "keyedXX" for the modules that I suppose are the most likely candidates that would need support for keyed nodes.

Ref commit e6d229c

**Comment by norpan** [comment]
Wow, that was fast :) Thanks!
Yes, it makes sense to have it only for those that are likely to be dynamic.

**Comment by rundis** [comment]
resolved in 2.0.0
